.gitignore

```diff
@@ -0,0 +1,3 @@
+/target
+**/*.rs.bk
+Cargo.lock
```

Comment:
Libraries should ignore Cargo.lock, only applications should track it.

https://github.com/rust-lang/cargo/issues/315

Reply:
While reproducible builds is not a must for libraries, keeping control of the dependencies should be. `Cargo.lock` helps in having a finer grained control of your dependencies. When upgrading the dependencies of your package you will not have a clear view of transient dependencies. Cargo.lock can help with that. It also helps in spotting if one of your dependencies has a license incompatible with Apache-2.0 which is something we should keep an eye on.

There are probably other ways to make sure that we do not increase the number of dependencies & that the license of dependencies is compatible, so if we find an alternative that can give us that without adding Cargo.lock to the repository I am fine with it.

Reply:
I understand why you want Cargo.lock, but it's not the policy for crates and I'd rather respect that, especially for a crate that has exactly 1 depedency (and I don't expect most initial rust-vmm crates to have much more than that).

Reply:
We already have 24 dependencies (transient + direct). Most of them are added by `tempfile`. This is a crate only used in tests. I guess we do not care about dev dependencies as they should not be included in the release binary. Cargo has a strange behaviour there because all dev dependencies are pulled even for a release build.

Reply:
Yes, they are pulled because you can do `cargo test --release`. But the only actual dependency of the library is libc.